### PR TITLE
[AD-197] Improve clarity of status bar info

### PR DESCRIPTION
**Pull request by goodsoft**
Status bar now looks like this:
```
 Syncing: 3.0% │ Local: block 46f3256c, epoch 18, slot 17279 │ Network: epoch 51, slot 12499
 Blockchain not synced yet, balances can be inaccurate
```

### CODE REVIEW

File: ariadne/app-qt/Glue.hs [hunk 1]
```diff
@@ -204,11 +205,14 @@ cardanoEventToUI = \case
   CardanoStatusUpdateEvent CardanoStatusUpdate{..} ->
     Just $ UiCardanoEvent $
       UiCardanoStatusUpdateEvent UiCardanoStatusUpdate
-        { tipHeaderHash = pretty tipHeaderHash
-        , tipSlot = pretty tipEpochOrSlot
-        , currentSlot = pretty currentSlot
-                     <> if isInaccurate then " (inaccurate)" else ""
+        { syncProgress = syncProgress >>= \frac -> return $ toFixed 1 (fromRational $ frac * 100) <> "%"
```
Comment (maksbotan) [inline ariadne/app-qt/Glue.hs]: Nitpick: is it the same as `toFixed 1 . fromRational . (*100) <$> syncProgress`?

File: ui/qt/src/Ariadne/UI/Qt/Widgets/StatusBar.hs [hunk 2]
```diff
@@ -31,25 +31,27 @@ initStatusBar :: QMainWindow.QMainWindow -> IO StatusBar
...
-  liftIO $ QLabel.setText tipHeaderHashLabel $ toString $
-    format ("Tip Hash: " % stext) tipHeaderHash
-  liftIO $ QLabel.setText tipSlotLabel $ toString $
-    format ("Tip slot: " % stext) tipSlot
-  liftIO $ QLabel.setText currentSlotLabel $ toString $
-    format ("Current slot: " % stext) currentSlot
+  StatusBar{..} <- ask
+  liftIO $ do
+    QLabel.setText blockchainLocalLabel $ toString $
+      format ("Local: " % stext) blockchainLocal
```
Comment (maksbotan) [inline ui/qt/src/Ariadne/UI/Qt/Widgets/StatusBar.hs]: Maybe while at it, ditch `format` from here and chage to `"Local: " <> blockchainLocal`, etc. I didn't know it's possible when I first wrote this.

File: ui/qt/src/Ariadne/UI/Qt/Widgets/StatusBar.hs [hunk 3]
```diff
@@ -31,25 +31,27 @@ initStatusBar :: QMainWindow.QMainWindow -> IO StatusBar
...
+    QLabel.setText blockchainNetworkLabel $ toString $
+      format ("Network: " % stext) blockchainNetwork
+    case syncProgress of
+      Just progress -> do
+        QLabel.setText syncProgressLabel $ toString $
+          format ("Syncing blockchain: " % stext % ". Balances can be inaccurate." ) progress
```
Comment (maksbotan) [inline ui/qt/src/Ariadne/UI/Qt/Widgets/StatusBar.hs]: I'd rather say "may be inaccurate".

File: ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs [hunk 4]
```diff
@@ -23,46 +23,48 @@ import Ariadne.UI.Vty.Face
...
-drawStatusWidget (Named ariadneURL) statusWidgetState =
+drawStatusWidget (Named ariadneURL) StatusWidgetState{..} =
   B.padTop (B.Pad 1) $
     B.updateAttrMap (B.mapAttrName "status" B.borderAttr) $
     B.withAttr "status" $
-    (case statusWidgetState ^. statusWidgetNewVersionL of
-      Just ver -> (B.<=> B.padLeftRight 1 (updateNotification ver))
-      Nothing -> identity
-    ) $
+    maybe identity (\ver -> (B.<=> updateNotification ver)) statusWidgetNewVersion $
```
Comment (maksbotan) [inline ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs]: `B.(<=>) . updateNotification` ?

File: ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs [hunk 5]
```diff
@@ -23,46 +23,48 @@ import Ariadne.UI.Vty.Face
...
+    maybe identity (\_ -> (B.<=> inaccurateNotification)) statusWidgetSyncProgress $
```
Comment (maksbotan) [inline ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs]: `const (B.<=> inaccurateNotification)` ?

File: ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs [hunk 6]
```diff
@@ -23,46 +23,48 @@ import Ariadne.UI.Vty.Face
...
-      , ("Tip slot", statusWidgetState ^. statusWidgetTipSlotL)
-      , ("Current slot", statusWidgetState ^. statusWidgetSlotL)
+      [ statusWidgetSyncProgress >>= \progress -> Just ("Syncing", progress)
+      , Just ("Local", statusWidgetBlockchainLocal)
+      , Just ("Network", statusWidgetBlockchainNetwork)
       ]
 
+    inaccurateNotification :: B.Widget name
+    inaccurateNotification = B.padLeftRight 1 . B.txt $
+      "Blockchain not synced yet, balances can be inaccurate"
```
Comment (maksbotan) [inline ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs]: Same for 'may'

File: ariadne/app-qt/Glue.hs (hunk 1 above)
Comment (goodsoft) [reply ariadne/app-qt/Glue.hs]: Uh-huh, `(<> "%") .`

File: ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs (hunk 4 above)
Comment (goodsoft) [reply ui/vty/src/Ariadne/UI/Vty/Widget/Status.hs]: Your code will put notification _above_ status bar (as first argument), not _below_ (as second argument)

**Pull request merged**